includes/components/class-attribute.php

```diff
@@ -307,6 +310,67 @@ public function register_models() {
 		}
 	}
 
+	/**
+	 * Validates unique model attributes.
+	 *
+	 * @param array  $errors Error messages.
+	 * @param object $model Model object.
+	 * @return array
+	 */
+	public function validate_unique_attributes( $errors, $model ) {
+
+		if ( ! empty( $errors ) ) {
+			return $errors;
+		}
+
+		// Set unique fields.
+		$fields = array_intersect_key(
+			array_filter(
+				$model->_get_fields(),
+				function ( $field ) {
+					return $field->get_value();
+				}
+			),
+			array_filter(
+				$this->get_attributes( $model::get_meta( 'name' ) ),
+				function ( $value ) {
+					return $value['unique'];
+				}
+			)
+		);
+
+		// Check fields.
+		if ( ! $fields ) {
+			return $errors;
+		}
+
+		foreach ( $fields as $field ) {
+
+			// Get models.
+			$models = $model->query()->filter(
+				[
+					'status__in' => [ 'auto-draft', 'draft', 'pending', 'publish' ],
+					'id__not_in' => [ $model->get_id() ],
+				]
+			)->set_args(
+				[
+					'meta_key'   => hp\prefix( $field->get_name() ),
```

Comment:
Please use the HivePress query API instead, since these fields are added as model fields (based on attributes), you can use the field_name=>field_value pairs in the filter method, but check if it works correctly both on the back-end (meta boxes) and front-end.